modules/document_repository/php/uploadfile.class.inc

```diff
@@ -0,0 +1,314 @@
+<?php
+/**
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Document_Repository
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\document_repository;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Document_Repository
+ * @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Uploadfile extends \NDB_Page
+{
+    public $skipTemplate = true;
+
+    /**
+     * Same as the permissions for the main menu filter page.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool true iff the user has access to this page.
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return (
+            $user->hasPermission('document_repository_delete')
+            || $user->hasPermission('document_repository_view')
+
+        );
+    }
+
+    /**
+     * Handle a post form
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        if ($request->getMethod() == "POST") {
```

Comment:
not a big deal, but it might be cleaner to use a switch rather than if / else if here.

Reply:
fixed